.github/workflows/selenium-e2e.yml

```diff
@@ -1,32 +1,48 @@
-# This is a basic workflow that is manually triggered
-
 name: Selenium end-to-end tests
 
-# Controls when the action will run. Workflow runs when manually triggered using the UI
-# or API.
 on:
   workflow_dispatch:
-    # Inputs the workflow accepts.
-    inputs:
-      name:
-        # Friendly description to be shown in the UI instead of 'name'
-        description: 'Person to greet'
-        # Default value if no value is explicitly provided
-        default: 'World'
-        # Input has to be provided for the workflow to run
-        required: true
-        # The data type of the input
-        type: string
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "greet"
-  greet:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
+  selenium-run:
+    name: Run Selenium E2E tests
+    runs-on: ubuntu-24.04
+    environment: dev
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-    # Runs a single command using the runners shell
-    - name: Send greeting
-      run: echo "Hello ${{ inputs.name }}"
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 23
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: '23'
+          cache: maven
+
+      - name: Install Chrome
+        run: |
+          set -eux
+          sudo apt-get update
+          sudo apt-get install -y wget fonts-liberation libasound2 libnss3 libgbm1 xdg-utils
+          wget -O /tmp/chrome.deb https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+          sudo dpkg -i /tmp/chrome.deb || sudo apt-get -f install -y
+          google-chrome --version
+          
+      - name: Create root-level config.properties (QA only)
+        env:
+          QA_CRM_URL: ${{ secrets.QA_CRM_URL }}
+          CRM_USERNAME: ${{ secrets.CRM_USERNAME }}
+          CRM_PASSWORD: ${{ secrets.CRM_PASSWORD }}
+        run: |
+          set -eux
+          cat > config.properties <<'EOL'
+          qa_crm_url='$QA_CRM_URL'
```

Comment:
Maybe drop the QA at the front of this secret if we're using environment secrets.